src/main/kotlin/org/rust/ide/surroundWith/expression/RustExpressionSurrounderBase.kt

```diff
@@ -20,7 +24,15 @@ abstract class RustExpressionSurrounderBase : Surrounder {
     }
 
     final override fun surroundElements(project: Project, editor: Editor, elements: Array<out PsiElement>): TextRange? {
-        return surroundExpression(project, editor, requireNotNull(targetExpr(elements)))
+        val expression = requireNotNull(targetExpr(elements))
+        val templateExpr = RustElementFactory.createExpression(project, getTemplate()) as RustExprElement
+
+        getLeafExpression(templateExpr).replace(expression)
+        val newExpression = expression.replace(templateExpr)
+
+        CodeInsightUtilCore.forcePsiPostprocessAndRestoreElement(newExpression)
+
+        return getSelectionRange(newExpression)
```

Comment:
I have mixed feelings on changing API semantics of `RustExpressionSurrounderBase`. Previous, low level, form might be useful for some complex surrounders in the future.

I'm curious what are @matklad's and @alexeykudinkin's opinions on this